File: component/src/main/java/com/mackwu/component/ui/EditTextActivity.java
package com.mackwu.component.ui;

import android.os.Bundle;
import android.view.inputmethod.EditorInfo;

import androidx.annotation.Nullable;

import com.mackwu.base.BaseActivity;
import com.mackwu.base.util.Logger;
import com.mackwu.base.viewmodel.BaseViewModel;
import com.mackwu.component.databinding.WidgetActivityEditTextBinding;
import com.mackwu.component.util.EditTextUtil;

/**
 * ===================================================
 * Created by MackWu on 2021/1/15 17:27
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public class EditTextActivity extends BaseActivity<BaseViewModel, WidgetActivityEditTextBinding> {

    @Override
    public void initView(@Nullable Bundle savedInstanceState) {

        // setOnEditorActionListener
        binding.etTest.setOnEditorActionListener((v, actionId, event) -> {
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                String text = binding.etTest.getText().toString();
                Logger.d("text=" + text);
                return true;
            }
            return false;
        });

        binding.btnShowSoftInput.setOnClickListener(v -> EditTextUtil.showSoftInput(binding.etTest));
        binding.btnHideSoftInput.setOnClickListener(v -> EditTextUtil.hideSoftInput( binding.etTest));
    }

}

File: view/src/main/java/com/mackwu/view/EditTextActivity.kt
package com.mackwu.view

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.Editable
import android.text.TextWatcher
import kotlinx.android.synthetic.main.activity_edit_text.*

/**
 * ===================================================
 * Created by MackWu on 2019/11/11 14:42
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
class EditTextActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_text)

        //
        et_test.addTextChangedListener(object : TextWatcher{
            override fun afterTextChanged(s: Editable?) {

            }

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
            }

            /**
             * @param s  输入框中改变后的字符串信息
             * @param start  输入框中改变后的字符串的起始位置
             * @param before 输入框中改变前的字符串的位置
             * @param count 输入框中改变后的一共输入字符串的数量
             */
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
            }

        })
    }
}